Source/NSOperation.m

```diff
@@ -533,6 +533,56 @@ - (void) _finish
 
 @end
 
+	  
+@implementation NSBlockOperation
+
+// Initialize
+- (id) init
+{
+  self = [super init];
+  if(self != nil)
+    {
+      _executionBlocks = [[NSMutableArray alloc] initWithCapacity: 10];
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_executionBlocks);
+  [super dealloc];
+}
+
+// Managing the blocks in the Operation
++ (instancetype)blockOperationWithBlock: (GSBlockOperationBlock)block
+{
+  NSBlockOperation *op = [[NSBlockOperation alloc] init];
```

Comment:
Using self here instead of NSBlockOperation would allow subclasses to reuse the code more easily. But then it is highly unlikely that there ever will be a subclass.

Reply:
Yep... I will change that.  Thanks.

Reply:
Done